src/main/java/io/stargate/sgv2/jsonapi/exception/catchable/UnsupportedCqlType.java

```diff
@@ -0,0 +1,21 @@
+package io.stargate.sgv2.jsonapi.exception.catchable;
+
+import com.datastax.oss.driver.api.core.metadata.schema.ColumnMetadata;
+
+/** Thrown when we encounter a Cql column definition that is not supported by the API. */
+public class UnsupportedCqlType extends CatchableApiException {
```

Comment:
This is used in the specific case where we are building an API view of the CQL table, so we can return the schema of the PK when doing an insert and later when reading. This is when we are trying to represent an existing CQL table inside of the API

 Your first example would be a missing codec exception, I think we do have some issues as you mention that it may not be that the type is not support - but that we do not know how to write to it - or that we will never write to it. 
 
I am going to rename to UnsupportedCqlTypeForDML and we can improve later as needed  when we get to handling DDL